staging/src/k8s.io/kubectl/pkg/cmd/version/skew_warning.go

```diff
@@ -0,0 +1,70 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package version
+
+import (
+	"fmt"
+	"io"
+	apimachineryversion "k8s.io/apimachinery/pkg/version"
+	"math"
+	"regexp"
+	"strconv"
+)
+
+// supportedMinorVersionSkew is the maximum supported difference between the client and server minor versions.
+// For example: client versions 1.18, 1.19, and 1.20 would be within the supported version skew for server version 1.19,
+// and server versions 1.18, 1.19, and 1.20 would be within the supported version skew for client version 1.19.
+const supportedMinorVersionSkew = 1
+
+func printVersionSkewWarning(w io.Writer, clientVersion, serverVersion apimachineryversion.Info) error {
+	if supported, err := isWithinSupportedVersionSkew(clientVersion, serverVersion); err != nil {
+		return err
+	} else if !supported {
+		fmt.Fprintf(w, "WARNING: version difference between client (%s.%s) and server (%s.%s) exceeds the supported minor version skew of +/-%d\n",
+			clientVersion.Major, clientVersion.Minor, serverVersion.Major, serverVersion.Minor, supportedMinorVersionSkew)
+	}
+	return nil
+}
+
+func isWithinSupportedVersionSkew(clientVersion, serverVersion apimachineryversion.Info) (bool, error) {
+	clientMajorVersion, clientMinorVersion, err := parseMajorMinorVersion(clientVersion)
+	if err != nil {
+		return false, err
+	}
+
+	serverMajorVersion, serverMinorVersion, err := parseMajorMinorVersion(serverVersion)
+	if err != nil {
+		return false, err
+	}
+
+	return clientMajorVersion == serverMajorVersion &&
+		math.Abs(float64(clientMinorVersion-serverMinorVersion)) <= supportedMinorVersionSkew, nil
+}
+
+func parseMajorMinorVersion(v apimachineryversion.Info) (int, int, error) {
```

Comment:
I am wondering if for re-use in different places (when required) parseMajorMinorVersion() should belong to version.go.

Reply:
Thanks for this comment. I found an existing function in API machinery to get the versions for comparison instead of writing my own.
https://github.com/kubernetes/kubernetes/blob/f170d33e1f320d7a3bcd2ae5a60b7d335ae382f6/staging/src/k8s.io/apimachinery/pkg/util/version/version.go#L111